R/plumber.R

```diff
@@ -332,12 +398,84 @@ plumber <- R6Class(
 
       private$mnts[[path]] <- router
     },
+    #' @description Register a hook
+    #' @param stage a character string. Point in the lifecycle of a request.
+    #' @param handler a hook function.
+    #' @details Plumber routers support the notion of "hooks" that can be registered
+    #' to execute some code at a particular point in the lifecycle of a request.
+    #' Plumber routers currently support four hooks:
+    #' \enumerate{
+    #'  \item `preroute(data, req, res)`
+    #'  \item `postroute(data, req, res, value)`
+    #'  \item `preserialize(data, req, res, value)`
+    #'  \item `postserialize(data, req, res, value)`
+    #' }
+    #' In all of the above you have access to a disposable environment in the `data`
+    #' parameter that is created as a temporary data store for each request. Hooks
+    #' can store temporary data in these hooks that can be reused by other hooks
+    #' processing this same request.\cr
+    #' \cr
+    #' One feature when defining hooks in Plumber routers is the ability to modify
+    #' the returned value. The convention for such hooks is: any function that accepts
+    #' a parameter named `value` is expected to return the new value. This could
+    #' be an unmodified version of the value that was passed in, or it could be a
+    #' mutated value. But in either case, if your hook accepts a parameter
+    #' named `value`, whatever your hook returns will be used as the new value
+    #' for the response.\cr
+    #' \cr
```

Comment:
```suggestion
    #' processing this same request.
    #'
    #' One feature when defining hooks in Plumber routers is the ability to modify
    #' the returned value. The convention for such hooks is: any function that accepts
    #' a parameter named `value` is expected to return the new value. This could
    #' be an unmodified version of the value that was passed in, or it could be a
    #' mutated value. But in either case, if your hook accepts a parameter
    #' named `value`, whatever your hook returns will be used as the new value
    #' for the response.
    #'
```